internal/iso_packaging/iso_packaging.go

```diff
@@ -0,0 +1,241 @@
+								}},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyNever,
+					Volumes: []v1.Volume{
+						{
+							Name: "config",
+							VolumeSource: v1.VolumeSource{
+								Projected: &v1.ProjectedVolumeSource{
+									Sources: []v1.VolumeProjection{{
+										ConfigMap: &v1.ConfigMapProjection{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: b.build.Spec.Kickstart.Name,
+											},
+										},
+									}},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if b.buildConfig.Spec.S3Service.AWS != nil {
+
+		config := b.buildConfig.Spec.S3Service.AWS
+
+		jobSpec.Spec.Template.Spec.Containers[0].EnvFrom = []v1.EnvFromSource{
+			{
+				SecretRef: &v1.SecretEnvSource{
+					LocalObjectReference: v1.LocalObjectReference{Name: config.CredsSecretReference.Name},
+					Optional:             aws.Bool(false)},
+			},
+		}
+		jobSpec.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+			{
+				Name:  "AWS_DEFAULT_REGION",
+				Value: config.Region,
+			},
+		}
+
+		jobSpec.Spec.Template.Spec.Containers[0].Command = append(
+			jobSpec.Spec.Template.Spec.Containers[0].Command,
+			gets3Target(config.Bucket, b.build))
+	}
+
+	if b.buildConfig.Spec.S3Service.GenericS3 != nil {
+		config := b.buildConfig.Spec.S3Service.GenericS3
+		jobSpec.Spec.Template.Spec.Containers[0].EnvFrom = []v1.EnvFromSource{
+			{
+				SecretRef: &v1.SecretEnvSource{
+					LocalObjectReference: v1.LocalObjectReference{Name: config.CredsSecretReference.Name},
+					Optional:             aws.Bool(false)},
+			},
+		}
+		jobSpec.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+			{
+				Name:  "AWS_DEFAULT_REGION",
+				Value: config.Region,
+			},
+			{
+				Name:  "AWS_ENDPOINT_URL",
+				Value: config.Endpoint,
+			},
+			{
+				Name:  "AWS_NO_VERIFY_SSL",
+				Value: fmt.Sprintf("%v", config.SkipSSLVerification),
+			},
+		}
+
+		if config.CABundleSecretReference != nil {
+			// From here: https://go.dev/src/crypto/x509/root_unix.go
+			jobSpec.Spec.Template.Spec.Containers[0].Env = append(
+				jobSpec.Spec.Template.Spec.Containers[0].Env,
+				v1.EnvVar{Name: "SSL_CERT_DIR", Value: "/cabundle"})
+
+			jobSpec.Spec.Template.Spec.Containers[0].VolumeMounts = append(
+				jobSpec.Spec.Template.Spec.Containers[0].VolumeMounts,
+				v1.VolumeMount{Name: "cabundle", ReadOnly: true, MountPath: "/cabundle/"})
+
+			jobSpec.Spec.Template.Spec.Volumes = append(
+				jobSpec.Spec.Template.Spec.Volumes,
+				v1.Volume{
+					Name: "cabundle",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName: config.CABundleSecretReference.Name,
+						},
+					},
+				})
+		}
+
+		jobSpec.Spec.Template.Spec.Containers[0].Command = append(
+			jobSpec.Spec.Template.Spec.Containers[0].Command,
+			gets3Target(config.Bucket, b.build))
+	}
+
+	err = b.client.Create(context.TODO(), jobSpec)
+	if err != nil {
+		return fmt.Errorf("Cannot applied job: %v", err)
+	}
+	b.jobSpec = jobSpec
```

Comment:
Do you really need to store the jobSpec? or just the name and namespace?

Reply:
at this point I would like to store, I do not think that have heavy implications like memory at this stage. but happy to remove.